arroyo/processing/strategies/run_task.py

```diff
@@ -21,21 +25,40 @@ def __init__(
     ) -> None:
         self.__function = function
         self.__next_step = next_step
+        # Invalid message offsets pending commit
+        self.__invalid_messages = InvalidMessageState()
+
+    def __forward_invalid_offsets(self) -> None:
+        if len(self.__invalid_messages):
+            self.__next_step.poll()
+            filter_msg = self.__invalid_messages.build()
+            if filter_msg:
+                try:
+                    self.__next_step.submit(filter_msg)
+                    self.__invalid_messages.reset()
+                except MessageRejected:
+                    pass
 
     def submit(
         self, message: Message[Union[FilteredPayload, TStrategyPayload]]
     ) -> None:
         if isinstance(message.payload, FilteredPayload):
             self.__next_step.submit(cast(Message[FilteredPayload], message))
         else:
-            result = self.__function(cast(Message[TStrategyPayload], message))
-            value = message.value.replace(result)
-            self.__next_step.submit(Message(value))
+            try:
+                result = self.__function(cast(Message[TStrategyPayload], message))
+                value = message.value.replace(result)
+                self.__next_step.submit(Message(value))
+            except InvalidMessage as e:
+                self.__invalid_messages.append(e)
+                raise e
```

Comment:
Yeah we do. If a function needs to be wrapped, we generally expect users will just use RunTask or one of those variants which will handle it

Reply:
I agree that we prefer standard strategies to be used. Though I would add a different perspective here, and you can decide. 
The DLQ is generally seen as a burden by application developers: new kafka topic, more code in the consumer, thinking about the process to recover messages. This in the past drove product teams to just catch invlid messages and throw them away instead of adopting a more sane and robust approach (DLQ). 
So reducing friction to adoption as much as possible should be seen as important here. That would create fewer reasons to take the shortcut.

Reply:
Lgtm. I personally prefer to keep this as internal API only too - not yet convinced this is a problem we should even be solving. Frankly the only cases this works for are those that already can easily use `RunTask` anyway (i.e. no concurrency)